MSMAll/MSMAllPipeline.sh

```diff
@@ -31,693 +32,232 @@
+opts_AddMandatory '--multirun-fix-concat-name' 'mrfixConcatName' 'rfMRI_REST' "if multi-run FIX was used, you must specify the concat name with this option"
+opts_AddMandatory '--multirun-fix-names-to-use' 'mrfixNamesToUse' 'rfMRI_REST1_LR@rfMRI_REST1_RL...' "MRfix run names to use, @-separated list, choose which runs should be used as resting state data"
+opts_AddMandatory '--output-fmri-name' 'OutputfMRIName' 'rfMRI_REST' "name to give to concatenated single subject scan"
+opts_AddMandatory '--high-pass' 'HighPass' 'integer' 'the high pass value that was used when running FIX' '--melodic-high-pass'
+opts_AddMandatory '--fmri-proc-string' 'fMRIProcSTRING' 'string' "file name component representing the preprocessing already done, e.g. '_Atlas_hp0_clean'"
+opts_AddMandatory '--msm-all-templates' 'MSMAllTemplates' 'string' "path to directory containing MSM All template files, e.g. 'YourFolder/global/templates/MSMAll'"
+opts_AddMandatory '--input-registration-name' 'InputRegName' 'MSMAll' "the registration string corresponding to the input files, e.g. 'MSMSulc'"
+opts_AddMandatory '--output-registration-name' 'OutputRegName' 'MSMAll' "the registration string corresponding to the output files, e.g. 'MSMAll_InitalReg"
+opts_AddMandatory '--high-res-mesh' 'HighResMesh' 'meshnum' "high resolution mesh node count (in thousands), like '164' for 164k_fs_LR"
+opts_AddMandatory '--low-res-mesh' 'LowResMesh' 'meshnum' "low resolution mesh node count (in thousands), like '32' for 32k_fs_LR"
+#MSMAll inputs
+opts_AddOptional '--module-name' 'ModuleName' 'string' "name of script or code used to run registration, defaults to 'MSMAll.sh'" 'MSMAll.sh'
+opts_AddOptional '--iteration-modes' 'IterationModes' 'string' "Specifieds what modalities:
+                            C=RSN Connectivity
+                            A=Myelin Architecture
+                            T=RSN Topography
+                          and number is the number of elements delimited by _
+                          So CA_CAT means one iteration using RSN Connectivity and Myelin
+                          Architecture, followed by another iteration using RSN Connectivity,
+                          Myelin Architecture, and RSN Topography. Defaults to 'CA_CAT'" 'CA_CAT'
+opts_AddOptional '--method' 'Method' 'string' "Possible values: DR, DRZ, DRN, WR, WRZ, WRN, defaults to 'WRN'" 'WRN'
+opts_AddOptional '--use-migp' 'UseMIGP' 'YES/NO' "whether to use MIGP (MELODIC's Incremental Group Principal Component Analysis), defaults to 'NO'" 'NO'
+opts_AddOptional '--ica-dim' 'ICAdim' 'integer' "ICA (Independent Component Analysis) dimension, defaults to '40'" '40'
+opts_AddOptional '--low-sica-dims' 'LowsICADims' 'num_num_num...' "the low sICA dimensionalities to use for determining weighting for individual projection, defaults to '7_8_9_10_11_12_13_14_15_16_17_18_19_20_21'" '7_8_9_10_11_12_13_14_15_16_17_18_19_20_21'
```

Comment:
I would use `,` or `@` as the separator - `@` is common in other pipelines, while `,` is more natural-looking than `_`.

Reply:
I agree. I kept this because the current `_` is used in the `MSMAll.sh` as the separator for inputs. I haven't planned to change the main `MSMAll.sh` script but maybe this one should also be updated. @glasserm Matt, am I right that we should also update the `MSMAll.sh`?

Reply:
I think the original reason for using _ was that the internal loop in `MSMAllPipeline.sh` was made to be able to run multiple registrations in sequence, and it did that by a nested set of delimiters (looks like `_`, `@`, and `+`).  That loop is now gone in this version (@glasserm, please take note).  If we don't need the loop functionality, then it would probably be good to simplify/standardize things like this (and perhaps other no longer needed abstractions, like `ModuleName`).

Reply:
I would rather not change this at this time.

Reply:
@glasserm by "not change this", do you mean the `_` separator, or the previous loop functionality (to run multiple registrations)?

It wouldn't be very hard to support more than one allowed separator character (if we still wanted the loop, separate arrays for each setting is another way to do that).